src/php/ext/grpc/channel_credentials.c

```diff
@@ -236,6 +236,47 @@ PHP_METHOD(ChannelCredentials, createInsecure) {
   RETURN_NULL();
 }
 
+/**
+ * Create XDS channel credentials
+ * @param ChannelCredentials|null $fallback_creds  The fallback credentials used
+ * if the channel target does not have the 'xds:///' scheme or if the xDS
+ * control plane does not provide information on how to fetch credentials
+ * dynamically.
+ * @return ChannelCredentials The xDS channel credentials object
+ */
+PHP_METHOD(ChannelCredentials, createXds) {
+  grpc_channel_credentials* xds_creds = NULL;
+  zval* fallback_creds = NULL;
+  if (zend_parse_parameters_ex(0,  // ZEND_PARSE_PARAMS_QUIET,
+                               ZEND_NUM_ARGS() TSRMLS_CC, "O!", &fallback_creds,
+                               grpc_ce_channel_credentials) != SUCCESS) {
+    zend_throw_exception(spl_ce_InvalidArgumentException,
+                         "createXds expects a fallback credentials",
+                         1 TSRMLS_CC);
+    return;
+  }
+  char* fallback_creds_hash_str = "";
+  if (!fallback_creds) {
+    grpc_channel_credentials* insecure_creds =
+        grpc_insecure_credentials_create();
+    xds_creds = grpc_xds_credentials_create(insecure_creds);
+  } else {
+    wrapped_grpc_channel_credentials* wrapped_fallback_creds =
+        PHP_GRPC_GET_WRAPPED_OBJECT(wrapped_grpc_channel_credentials,
+                                    fallback_creds);
+    xds_creds = grpc_xds_credentials_create(wrapped_fallback_creds->wrapped);
+    fallback_creds_hash_str = wrapped_fallback_creds->hashstr;
+  }
+
+  // prefix "XDS:" as the hash of the xDS channel
```

Comment:
Not sure I understand this. `XDS:` scheme is used in the target string and only when user has used it we use the xDS resolver and xDS supplied creds. I don't understand using it here before the `hash_str`. May be there is a  design doc that explains this?

Reply:
This is an implementation details for the php extension that is irrelevant to the C-core. This is only used in the PHP extension layer to calculate a hash so that we can persist this object in the PHP extension. This does not get passed to the c-core in any way.

Reply:
ok